docs/drivers/azure.md

```diff
@@ -10,51 +10,105 @@ parent="smn_machine_drivers"
 
 # Microsoft Azure
 
-Create machines on [Microsoft Azure](http://azure.microsoft.com/).
+You will need an Azure Subscription to use this Docker Machine driver.
+[Sign up for a free trial.][trial]
 
-You need to create a subscription with a cert. Run these commands and answer the questions:
+> **NOTE:** This documentation is for the new version of  Azure driver started
+> shipping with v0.7.0 and it is not backwards-compatible with the old Azure
+> driver. If you like to manage your existing Azure machines, please download
+> and use machine versions prior to v0.7.0.
 
-    $ openssl req -x509 -nodes -days 365 -newkey rsa:1024 -keyout mycert.pem -out mycert.pem
-    $ openssl pkcs12 -export -out mycert.pfx -in mycert.pem -name "My Certificate"
-    $ openssl x509 -inform pem -in mycert.pem -outform der -out mycert.cer
+[azure]: http://azure.microsoft.com/
+[trial]: https://azure.microsoft.com/free/
 
-Go to the Azure portal, go to the "Settings" page (you can find the link at the bottom of the
-left sidebar - you need to scroll), then "Management Certificates" and upload `mycert.cer`.
+## Authentication
 
-Grab your subscription ID from the portal, then run `docker-machine create` with these details:
+First time you try to create a machine, Azure driver will ask you to
+authenticate:
 
-    $ docker-machine create -d azure --azure-subscription-id="SUB_ID" --azure-subscription-cert="mycert.pem" A-VERY-UNIQUE-NAME
+    $ docker-machine create --driver azure <machine-name>
+    Running pre-create checks...
+    Microsoft Azure: To sign in, use a web browser to open the page https://aka.ms/devicelogin.
+    Enter the code [...] to authenticate.
 
-The Azure driver uses the `b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-15_10-amd64-server-20151116.1-en-us-30GB`
-image by default. Note, this image is not available in the Chinese regions. In China you should
- specify `b549f4301d0b4295b8e76ceb65df47d4__Ubuntu-15_10-amd64-server-20151116.1-en-us-30GB`.
+After authenticating, the driver will remember your credentials up to two weeks.
```

Comment:
It will be good to add a comment saying that, if someone wants to automate using docker-machine, they will have to provide publishsettings or subscriptionID, cert pair